src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -313,7 +304,7 @@ public async Task RefreshAsync(CancellationToken cancellationToken)
                                 ConfigurationSetting setting = change.Current;
                                 foreach (Func<ConfigurationSetting, ValueTask<ConfigurationSetting>> func in _options.UserDefinedMappers)
                                 {
-                                    setting = func(setting).Result;
+                                    setting = await func(setting);

```

Comment:
`await` should be accompanied by `.ConfigureAwait(false)`.

`setting = await func(setting).ConfigureAwait(false);`